src/Features/CSharp/Portable/CodeLens/CSharpCodeLensDisplayInfoService.cs

```diff
@@ -0,0 +1,202 @@
+                .WithTypeQualificationStyle(SymbolDisplayTypeQualificationStyle.NameAndContainingTypes);
+
+        /// <summary>
+        /// Returns the node that should be displayed
+        /// </summary>
+        public SyntaxNode GetDisplayNode(SyntaxNode node)
+        {
+            while (true)
+            {
+                switch (node.Kind())
+                {
+                    // LocalDeclarations do not have symbols themselves, you need a variable declarator
+                    case SyntaxKind.LocalDeclarationStatement:
+                        var localDeclarationNode = (LocalDeclarationStatementSyntax) node;
+                        node = localDeclarationNode.Declaration.Variables.First();
+                        continue;
+
+                        // Field and event declarations do not have symbols themselves, you need a variable declarator
+                    case SyntaxKind.FieldDeclaration:
+                    case SyntaxKind.EventFieldDeclaration:
+                        var fieldNode = (BaseFieldDeclarationSyntax) node;
+                        node = fieldNode.Declaration.Variables.First();
+                        continue;
+
+                        // Variable is a field without access modifier. Parent is FieldDeclaration
+                    case SyntaxKind.VariableDeclaration:
+                        node = node.Parent;
+                        continue;
+
+                        // Built in types   
+                    case SyntaxKind.PredefinedType:
+                        node = node.Parent;
+                        continue;
+
+                    case SyntaxKind.MultiLineDocumentationCommentTrivia:
+                    case SyntaxKind.SingleLineDocumentationCommentTrivia:
+                        // For DocumentationCommentTrivia node, node.Parent is null. Obtain parent through ParentTrivia.Token
+                        if (node.IsStructuredTrivia)
+                        {
+                            var structuredTriviaSyntax = (StructuredTriviaSyntax) node;
+                            node = structuredTriviaSyntax.ParentTrivia.Token.Parent;
+                            continue;
+                        }
+                        return null;
+
+                    default:
+                        return node;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the DisplayName for the given node.
+        /// </summary>
+        public string GetDisplayName(SemanticModel semanticModel, SyntaxNode node, bool useShortName)
+        {
+            if (node == null)
+            {
+                return CSharpFeaturesResources.paren_unknown_paren;
+            }
+
+            var symbolDisplayFormat = useShortName ? ShortFormat : DefaultFormat;
+            string displayName;
+            string enclosingScopeString;
+
+            int lastDotBeforeSpaceIndex;
+
+            ISymbol symbol;
+
+            if (SyntaxFacts.IsGlobalAttribute(node))
```

Comment:
Don't see why this needs to be in the compiler layer.  It could totally be in the workspace layer afaict.